src/sentry/rules/conditions/event_frequency.py

```diff
@@ -261,15 +261,24 @@ def disable_consistent_snuba_mode(
         return option_override_cm
 
     def get_comparison_start_end(
-        self, interval: timedelta, duration: timedelta
+        self, end: datetime, duration: timedelta, comparison_interval: timedelta | None = None
     ) -> tuple[datetime, datetime]:
         """
-        Calculate the start and end times for the query. `interval` is only used for EventFrequencyPercentCondition
-        as the '5 minutes' in The issue affects more than 100 percent of sessions in 5 minutes, otherwise it's the current time.
-        `duration` is the time frame in which the condition is measuring counts, e.g. the '10 minutes' in
-        "The issue is seen more than 100 times in 10 minutes"
+        Calculate the start and end times for the query.
+        Take these example conditions that generate queries:
+            '# of issues in 1 day is {}'
+            "# of issues is {}% higher in 1 day compared to 12 hours ago"
+
+        "duration" is the period of time we're querying over, in this case 1 day
+        for both queries.
+        "comparison_interval" only applies for percent comparison queries (the
+        second condition) and is the time diff between the two queries we derive a
```

Comment:
> diff between the two queries we derive a percentage from
🤔 this is a bit confusing and the example isn't really helping me much here...

I think what you're trying to say is that while the duration is 24hrs, we've offset our comparison by 12hrs. So the query becomes comparing [24hrs duration from now] to [24hrs duration from 12hrs ago]?

Reply:
I can cleanup the explanation